internal/validator/notation/trust_store.go

```diff
@@ -0,0 +1,91 @@
+package notation
+
+import (
+	"connaisseur/internal/validator/auth"
+	"context"
+	"crypto/x509"
+	"encoding/pem"
+	"fmt"
+
+	"github.com/notaryproject/notation-go/verifier/truststore"
+)
+
+type TrustStoreKey struct {
+	name  string
+	type_ truststore.Type
+}
+
+type InMemoryTrustStore struct {
+	trustRoots []auth.TrustRoot
+	certs      map[TrustStoreKey][]*x509.Certificate
+	truststore.X509TrustStore
+}
+
+func NewInMemoryTrustStore(trustRoots []auth.TrustRoot) (*InMemoryTrustStore, error) {
+	certs := make(map[TrustStoreKey][]*x509.Certificate)
+
+	for _, trustRoot := range trustRoots {
+		if trustRoot.Cert == "" {
+			return &InMemoryTrustStore{}, fmt.Errorf(
+				"no certificate provided for trust root %s",
+				trustRoot.Name,
+			)
+		}
+
+		cert, err := parseCertificate(trustRoot.Cert)
+		if err != nil {
+			return &InMemoryTrustStore{}, fmt.Errorf(
+				"failed to parse certificate for trust root %s: %w",
+				trustRoot.Name,
+				err,
+			)
+		}
+		certs[TrustStoreKey{name: trustRoot.Name, type_: truststore.TypeCA}] = cert
+
+		// add timestamp cert, if present
+		if trustRoot.TSCert != "" {
+			tsCert, err := parseCertificate(trustRoot.TSCert)
+			if err != nil {
+				return &InMemoryTrustStore{}, fmt.Errorf(
+					"failed to parse timestamp certificate for trust root %s: %w",
+					trustRoot.Name,
+					err,
+				)
+			}
+			certs[TrustStoreKey{name: trustRoot.Name, type_: truststore.TypeTSA}] = tsCert
+		}
+	}
+
+	return &InMemoryTrustStore{
+		trustRoots: trustRoots,
+		certs:      certs,
+	}, nil
+}
+
+func parseCertificate(cert string) ([]*x509.Certificate, error) {
```

Comment:
nit: name and return type don't really match. maybe just return the cert and build the list in the outer function?